network-unix.c

```diff
@@ -194,13 +194,19 @@ int do_select(int fd, unsigned int timeout)
 {
 	struct pollfd pfd;
 	int ret;
+	int poll_timeout;
 
 	pfd.fd = fd;
 	pfd.events = POLLOUT | POLLERR;
 	pfd.revents = 0;
 
+	if (!timeout)
```

Comment:
But can we get a 0 value though? Looking at the code it seems we always call this with `NETWORK_TIMEOUT_MS`?

Reply:
Yes, there is one case (that I know of) where we can get the 0 value and it's when creating a network context.
`do_select()` can be called by `do_connect()` (network.c) which is called by `create_socket()` (network.c) which can also be called by `network_create_context()` (network.c) which is called by `iio_create_context()` (context.c) via `ops->create()`.